symengine/tests/expression/test_expression.cpp

```diff
@@ -0,0 +1,40 @@
+#include "catch.hpp"
+#include <chrono>
+#include <iostream>
+
+#include <symengine/symengine_rcp.h>
+#include <symengine/expression.h>
+#include <symengine/pow.h>
+#include <symengine/symbol.h>
+
+using namespace SymEngine;
+
+TEST_CASE("Constructors of expression", "[expression]")
+{
+	expression e0 = new Symbol("x");
+}
+
+TEST_CASE("Printing of expression", "[expression]")
+{
+	expression e0 = new Symbol("x");
```

Comment:
Just pass it a symbol("x"), it looks like it will just work.

Sent from my mobile phone.
On Jul 4, 2015 11:19 AM, "Ondřej Čertík" <ondrej.certik@gmail.com> wrote:

> Yes. We need to probably make the expression accept rcp<basic>, as usual.
>
> Sent from my mobile phone.
> On Jul 4, 2015 9:38 AM, "Sumith" <notifications@github.com> wrote:
>
>> In symengine/tests/expression/test_expression.cpp
>> <https://github.com/sympy/symengine/pull/500#discussion_r33888287>:
>>
>> > +
>> > +#include <symengine/symengine_rcp.h>
>> > +#include <symengine/expression.h>
>> > +#include <symengine/pow.h>
>> > +#include <symengine/symbol.h>
>> > +
>> > +using namespace SymEngine;
>> > +
>> > +TEST_CASE("Constructors of expression", "[expression]")
>> > +{
>> > +	expression e0 = new Symbol("x");
>> > +}
>> > +
>> > +TEST_CASE("Printing of expression", "[expression]")
>> > +{
>> > +	expression e0 = new Symbol("x");
>>
>> Is using new a problem?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/sympy/symengine/pull/500/files#r33888287>.
>>
>